attribution/experiment_logger.py

```diff
@@ -173,14 +176,126 @@ def log_perturbation(
         }
 
     def clean_tokens(self, tokens):
-        return [token.replace("Ġ", "") for token in tokens]
+        return [token.replace("Ġ", " ") for token in tokens]
 
-    def print_sentence_attribution(self, score_agg: Literal["mean", "sum", "last"] = "mean"):
-        sentences = []
+    def score_to_color(self, score, vmin=-1, vmax=1):
+        norm = mcolors.Normalize(vmin=vmin, vmax=vmax)
+        cmap = plt.cm.coolwarm
+        rgba_color = cmap(norm(score))
+        color_hex = mcolors.to_hex(rgba_color)
+        return color_hex
 
-        for (exp_id, attr_strat), exp_data in self.df_input_token_attribution.groupby(
-            ["exp_id", "attribution_strategy"]
-        ):
+    def print_text_total_attribution(
+        self, exp_id: Optional[int] = None, score_agg: Literal["mean", "sum", "last"] = "mean"
+    ):
+        if exp_id == -1:
+            exp_id = self.df_experiments["exp_id"].max()
+
+        token_attrs_df = (
+            self.df_input_token_attribution.groupby(["exp_id", "attribution_strategy"])
+            if exp_id is None
+            else self.df_input_token_attribution[
+                self.df_input_token_attribution["exp_id"] == exp_id
+            ].groupby(["exp_id", "attribution_strategy"])
+        )
+
+        for (exp_id, attr_strat), exp_data in token_attrs_df:
```

Comment:
We can maybe do `for (exp_id, _), ...` if `attr_strat` is not used

Reply:
updated :)